### ISSUE TITLE
Action, Task, and Pipeline names cannot contain quotes and slashes

### ISSUE THREAD

**Issue by jrobble** [openmpf/openmpf#481]
Using quotes will cause the following code in the merge_aatp() function in customPipelines.py to fail:

```
    # Verify the root tags match
    if installed_xml_data_root.tag == custom_xml_data_root.tag:
        # For each custom item
        for item in custom_xml_data_children:

            # Check if the custom item already exists in the installed file. If not, add it.
            if not installed_xml_data_root.findall("{0}[@name='{1}']".format(item.tag, item.get('name'))):
                # Add the custom items to the XML tree under root element
                installed_xml_data_root.append(item)
```

Specifically, the problem is with the following predicate:

```
"{0}[@name='{1}']".format(item.tag, item.get('name')
```

If customActions.xml contains:

```
<?xml version='1.0' encoding='UTF-8'?>
<actions>
  <action name="CUSTOM JEFF'S FACECV ACTION" algorithm-ref="FACECV">
    <description>JEFF'S FACECV</description>
    <property-refs>
      <property-ref name="FRAME_INTERVAL">
        <value>77</value>
      </property-ref>
    </property-refs>
  </action>
</actions>
```

The apostrophe (single quote) in "JEFF'S" will cause the predicate string to be `action[@name='CUSTOM JEFF'S FACECV ACTION']`. Notice that the apostrophe interferes with the quoting of the content after `name=`.

According to the XPath syntax [documentation](https://docs.python.org/3/library/xml.etree.elementtree.html#elementtree-xpath), quotes are not allowed in that context.

Note that customPipelines.py is used during an OpenMPF upgrade procedure. 

To avoid this problem, users should not be allowed to create pipelines with single or double quotes through the pipeline creation tool UI.

**Disable any special characters but _ and -**

**Comment by dpierce17** [comment]
Can you provide a hint where I would find the entry point to the pipeline creation tool UI in the WFM? I.e. which class should I start looking in?

**Comment by jrobble** [comment]
From the top menu, click "Configuration" --> "Pipelines". Also, please make sure that "Configuration" --> "Pipelines 2" doesn't allow the special characters either.

Please refer to the [User Guide](https://openmpf.github.io/docs/site/User-Guide/index.html#create-custom-pipelines) to learn how to use the "Pipelines" page. The "Pipelines 2" page is experimental, so there's no guide. I can provide further assistance with that if you need it.

Also, I realized that the following part of the User Guide is incorrect:

> Selecting "Create Custom Pipelines" from the Workflow Manager menu brings up the Pipeline Creation View

I will fix it.

Start by looking at `PipelineController.addPipeline()`.

**Comment by dpierce17** [comment]
Is it ok to use a whitelist, i.e. pipeline name can only contain letters, numbers, spaces, dashes or underscores?

**Comment by jrobble** [comment]
A whitelist sounds safer than a blacklist. Works for me.

**Comment by dpierce17** [comment]
I have the whitelist working to check the pipelineName in both the WFM Configuration Pipeline and Pipeline2 options. But, the javascript software that calls the pipeline endpoints don't handle the WfmProcessingException that is being thrown in the PipelineController very well. A generic error is being displayed to the user, but it doesn't contain reason for the error that is included in the WfmProcessingException. Should I make fixing the javascript (PipelinesCtrl.js and Pipelines2Ctrl.js) part of this task or is fixing the UI an over-reach for this issue?

**Comment by jrobble** [comment]
Please generate another task for fixing the JavaScript. Let's leave it for another time.

**Comment by dpierce17** [comment]
Landed with PR#498
